### PR TITLE
Implement Union Type

### PR DESCRIPTION
**Pull request by markolson**

### CODE REVIEW

File: lib/graphql/type/introspection.ex [hunk 1]
```diff
@@ -126,7 +125,7 @@ defmodule GraphQL.Type.Introspection do
               thunk_fields = GraphQL.Execution.Executor.maybe_unwrap(schema.fields)
               Enum.map(thunk_fields, fn({n, v}) -> Map.put(v, :name, n) end)
             (s, _, _) ->
-              []
+              nil
```
Comment (aweiker) [inline lib/graphql/type/introspection.ex:128]: was this done to match the spec?

Comment (markolson) [inline lib/graphql/type/introspection.ex:128]: Yes, or at least the js reference implementation's results.

File: test/graphql/type/union_interface_test.exs [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+          ],
+          "pets" => [
+            %{"__typename" => "Dog", "barks" => true, "name" => "Odie"},
+            %{"__typename" => "Cat", "meows" => false, "name" => "Garfield"}
+          ]
+        }
+  end
+
+  test "gets execution info in resolver" do
+
```
Comment (aweiker) [inline test/graphql/type/union_interface_test.exs:277]: Empty test here
Here's the reference from the JS implementation. https://github.com/graphql/graphql-js/blob/042e1620bb3a83e07e15586bb5b69b211c91e8de/src/execution/__tests__/union-interface.js#L350

**Comment by joshprice** [comment]
:+1: Looks good, reviewed with @freshtonic

**Pull request merged**